improver_tests/blending/weighted_blend/test_WeightedBlendAcrossWholeDimension.py

```diff
@@ -473,6 +516,21 @@ def test_with_weights(self):
         self.assertIsInstance(result, iris.cube.Cube)
         self.assertArrayAlmostEqual(result.data, BLENDED_PERCENTILE_DATA)
 
+    @ManageWarnings(ignored_messages=[COORD_COLLAPSE_WARNING])
+    def test_with_weights_perc_as_float64(self):
+        """Test function when a data cube and a weights cube are provided. But with
+        a percentile cood that is float64"""
+
+        perc_cube = percentile_cube()
+        coord = "forecast_reference_time"
+        plugin = WeightedBlendAcrossWholeDimension(coord)
+        perc_coord = perc_cube.coord("percentile")
+        perc_coord.points = perc_coord.points.astype(np.float64)
+        result = plugin.percentile_weighted_mean(perc_cube, self.weights1d, perc_coord)
+        self.assertIsInstance(result, iris.cube.Cube)
+        self.assertArrayAlmostEqual(result.data, BLENDED_PERCENTILE_DATA)
+        self.assertEqual(result.coord("percentile").points.dtype, np.float32)
```

Comment:
Why is that?

Reply:
Well, we have a blanket ban on using float64 within the code, and I think if the data's loaded in, it's converted to float32 anyway (as shown here). I'm just not sure that anything would actually break if it wasn't converted or, indeed if it is. In which case testing for something that won't break the code feels a little superfluous.

Reply:
Then should the lines from `improver/blending/weighted_blend.py` be removed.
```python
        # Ensure collapsed coordinates do not promote themselves
        # to float64.
        for coord in cube_new.coords():
            if coord.points.dtype == np.float64:
                coord.points = coord.points.astype(np.float32)
```

Reply:
No, it's actually because of these line that I don't think we need to test it, the data is always converted. If you were to delete the lines, then I guess we'd need to test that the coordinates weren't using float64, but with an explicit type conversion here, I'm just not sure it's necessary.

It won't break anything to keep it in, it just feels superfluous to me, perhaps get a second opinion :-)

Reply:
Generally, I think it's OK to skip explicit testing of float64 inputs where the intention is for all data to be converted to float32 anyway. That'll mean a small amount of un-covered lines, but I'd prefer for the unit tests to focus on "what parts of the code are likely to contain bugs and unpredictable/sneaky behaviour" rather than "need to get to 100% line coverage to make the metrics happy".

In this case where the test has already been written, I think leave it as-is rather than making another commit to drop the test.